continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -856,14 +852,31 @@ object DefDefTransforms extends TreesChecks:
                 continuationsStateMachineLabelSelect,
                 continuationsStateMachineLabelSelect.select(integerOR).appliedTo(integerMin)
               )
-            ),
-            ref(transformedMethod.symbol).appliedToTermArgs(
-              List.fill(numTransformedMethodOriginalParams)(nullLiteral) :+
-                continuationsStateMachineThis
-            )
+            ), {
+              val methodSymbol = transformedMethodSymbol
+              println(s"methodSymbol.paramSymss: ${methodSymbol.paramSymss}")
+              methodSymbol
+                .paramSymss
+                .foldLeft(Option.empty[tpd.Tree]) { (acc, nextArgs) =>
+                  if (nextArgs.exists(_.isType))
+                    val nextArgsAsNothing = nextArgs.map(_ => ctx.definitions.NothingType)
+                    acc
+                      .map(_.appliedToTypes(nextArgsAsNothing))
+                      .orElse(Some(ref(methodSymbol).appliedToTypes(nextArgsAsNothing)))
+                  else
+                    val args = nextArgs.map { sym =>
+                      if (sym.info.hasClassSymbol(requiredClass(continuationFullName))) {
+                        continuationsStateMachineThis
+                      } else nullLiteral
+                    }
+                    acc
+                      .map(_.appliedToArgs(args))
+                      .orElse(Some(ref(methodSymbol).appliedToArgs(args)))
```

Comment:
Perhaps this can split the composition of the `acc` with getting the next args... 

```suggestion
                  
                  val applyArgs = if (nextArgs.exists(_.isType))
                    nextArgs.map(_ => ctx.definitions.NothingType)
                  else
                    nextArgs.map { sym =>
                      if (sym.info.hasClassSymbol(requiredClass(continuationFullName)))
                        continuationsStateMachineThis
                      else nullLiteral
                    }

                  val operator = acc.getOrElse(ref(methodSymbol))
                  Some(operator.appliedToArgs(applyArgs))
```